cmd/juno/juno.go

```diff
@@ -381,10 +391,99 @@ func GenP2PKeyPair() *cobra.Command {
 
 			pubHex := make([]byte, hex.EncodedLen(len(rawPub)))
 			hex.Encode(pubHex, rawPub)
-			fmt.Println("P2P Public Key:", string(pubHex))
+			_, err = fmt.Fprintln(cmd.OutOrStdout(), "P2P Public Key:", string(pubHex))
+			if err != nil {
+				return err
+			}
+
+			_, err = fmt.Fprintln(cmd.OutOrStdout(), "P2P PeerID:", id)
+			if err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+}
+
+//nolint:gocyclo
+func DBSize() *cobra.Command {
+	dbSizeCmd := &cobra.Command{
+		Use:   "db-size",
+		Short: "Calculate's Juno's DB size.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			dbPath, err := cmd.Flags().GetString(dbPathF)
+			if err != nil {
+				return err
+			}
+
+			if dbPath == "" {
+				return fmt.Errorf("--%v cannot be empty", dbPathF)
+			}
+
+			pebbleDB, err := pebble.New(dbPath)
+			if err != nil {
+				return err
+			}
+
+			var totalSize, stateSizeWithoutHistory, stateSizeWithHistory uint
+
+			_, err = fmt.Fprintln(cmd.OutOrStdout(), "Total number of DB buckets:", uint(db.SchemaIntermediateState)+1)
+			if err != nil {
+				return err
+			}
+
+			var bucketSize uint
+			for i := db.StateTrie; i <= db.SchemaIntermediateState; i++ {
+				bucketSize, err = pebble.CalculatePrefixSize(cmd.Context(), pebbleDB.(*pebble.DB), []byte{byte(i)})
+				if err != nil {
+					return err
+				}
+
+				_, err = fmt.Fprintln(cmd.OutOrStdout(), uint(i)+1, "Size of", i, "=", bucketSize)
+				if err != nil {
+					return err
+				}
+
+				totalSize += bucketSize
+
+				if i == db.StateTrie || i == db.ContractStorage || i == db.Class || i == db.ContractNonce || i == db.ClassesTrie ||
```

Comment:
What do you think about defining a map with these variables, and just checking if i exists in the map?

Reply:
same for below

Reply:
Nice I will do that

Reply:
I have done something like in the utils/slice

```go
func Contains[T comparable](slice []T, e T) bool {
	for _, v := range slice {
		if e == v {
			return true
		}
	}
	return false
}
```

Reply:
there is a function with same name in stdlib: https://pkg.go.dev/slices#Contains

Reply:
I proposed AnyOf because the stdlib version lacks a variadic parameter

Reply:
Done

Reply:
Just saw your comment

Reply:
Changed the Contains code to Anyof